### PR TITLE
Export clj-kondo hook

### PR DESCRIPTION
**Pull request by mainej**
This exports the clj-kondo hook for `manifold.deferred/both`, along with the other clj-kondo config used internally by manifold.

## Background

After upgrading my project's deps from manifold `0.2.4` to `0.4.2`, I started getting unresolved-var warnings from clj-kondo everywhere I use any function that's defined after [this line](https://github.com/clj-commons/manifold/blob/12950654f185c65156a5817f5bec79a3afbfe401/src/manifold/deferred.clj#L466), which uses the `both` macro.

The warnings went away after merging the config from `.clj-kondo/config.edn`, which defines a hook that teaches clj-kondo how to interpret `both`, but which is available only while hacking on manifold itself, into the config in `resources/clj-kondo.exports/manifold/manifold/config.edn`, which is exported to projects that depend on manifold.

There's no change to hacking on manifold itself, because the internal config includes the exported config via `:config-paths`. This seems better than maintaining the clj-kondo config in two places.

### CODE REVIEW

File: .clj-kondo/config.edn [hunk 1]
```diff
@@ -1,20 +1 @@
...
-                 {:linters {:redefined-var {:level :off}}}
-
-                 manifold.stream.core/def-sink
-                 {:linters {:redefined-var {:level :off}}}
-
-                 manifold.stream.core/def-source
-                 {:linters {:redefined-var {:level :off}}}}
-
- :config-paths ["../resources/clj-kondo.exports/manifold/manifold/"]}
+{:config-paths ["../resources/clj-kondo.exports/manifold/manifold/"]}
```
Comment (ferdinand-beyer) [inline .clj-kondo/config.edn:1]: I think there is one `manifold/` directory too many?

**Comment by KingMob** [comment]
@mainej Thanks for this, this has sort of been on my to-do list forever. Aleph and Manifold are pretty macro-heavy (well, for Clojure), and there's some funky stuff going on, too, so clj-kondo is extra-helpful.

I'll check that it works this weekend.

File: .clj-kondo/config.edn (hunk 1 above)
Comment (mainej) [reply .clj-kondo/config.edn:1]: When clj-kondo is copying files out of the jar, it [looks like](https://github.com/clj-kondo/clj-kondo/blob/0a3f6837fa0964a1035cf8da05836bc8444d4279/src/clj_kondo/impl/core.clj#L220-L230) it doesn't really care what the subdirectories of `clj-kondo.exports/` are called, it just wants them to be two layers deep. I think it ignores the first layer ( `clj-kondo.exports/manifold` or `clj-kondo.exports/clj-commons` or whatever), and copies the second (`clj-kondo.exports/whatever/manifold`) into the project's `.clj-kondo/`, ending up with `.clj-kondo/manifold/config.edn`. I'm not exactly sure how it then later uses those copied configs, but things seem to work out OK.

Comment (KingMob) [reply .clj-kondo/config.edn:1]: I love kondo, but the docs on this aspect have always been a bit confusing.

Comment (ferdinand-beyer) [reply .clj-kondo/config.edn:1]: `clj-kondo` [recommends this convention][docs]:

```
clj-kondo.exports/<your-org>/<your-libname>/
```

It seems reasonable to require a quasi-unique namespace for `<your-org>`, similar to maven dependencies (e.g. Clojars).

[docs]: https://github.com/clj-kondo/clj-kondo/blob/master/doc/config.md#exporting

Comment (KingMob) [reply .clj-kondo/config.edn:1]: It's not that part, but the part about simultaneously using, and exporting, the same files.

**Comment by KingMob** [comment]
Ehh, this weekend was a bust. Life happened.

@ferdinand-beyer, does it work for you? If you want to approve it, we can merge. It's fairly low-risk.

**Comment by ferdinand-beyer** [comment]
> Ehh, this weekend was a bust. Life happened.
> 
> @ferdinand-beyer, does it work for you? If you want to approve it, we can merge. It's fairly low-risk.

Yeah, life tends to get in the way :)

Sure, works for me. Let's merge this. Thanks!

**Review by ferdinand-beyer — APPROVED** [review]

**Review by arnaudgeiser — APPROVED** [review]

**Pull request merged**